lib/LaTeXML/Package/TeX.pool.ltxml

```diff
@@ -2238,9 +2239,7 @@ DefPrimitive('\read Number SkipKeyword:to SkipSpaces Token', sub {
         push(@tokens, $t);
         $level++ if $cc == CC_BEGIN;
         $level-- if $cc == CC_END;
-        last if ((($cc == CC_SPACE) && ($t->getString eq "\n"))
-          || ($cc == CC_COMMENT)
-          || ($t->equals(T_CS('\par')))) && !$level; }
+        last if $mouth->isEOL && !$level; }
```

Comment:
you may find this to be a silly optimization but `!$level && $mouth->isEOL` is bound to be faster in the cases $level isn't  set -- no need to call the subroutine.

Reply:
and certainly can't hurt; will fix.